gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@8.11.3 | darwin | x64
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/ws/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/ken/.node-gyp/8.11.3/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/ken/.node-gyp/8.11.3',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/ken/.node-gyp/8.11.3/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/ws',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
In file included from ../src/bufferutil.cc:15:
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:213:5: error: no member named 'ThrowException' in namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
      ~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:213:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:42: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                             ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:218:9: error: no type named 'ThrowException' in namespace 'v8'
    v8::ThrowException(error);
    ~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:222:65: error: no member named 'New' in 'v8::String'
    v8::Local<v8::Value> err = v8::Exception::Error(v8::String::New(msg));
                                                    ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:223:50: error: expected '(' for function-style cast or type construction
    v8::Local<v8::Object> obj = err.As<v8::Object>();
                                       ~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:223:52: error: expected expression
    v8::Local<v8::Object> obj = err.As<v8::Object>();
                                                   ^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:224:65: error: too few arguments to function call, expected 2, have 1
    obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
                                      ~~~~~~~~~~~~~~            ^
/Users/ken/.node-gyp/8.11.3/include/node/v8.h:2929:3: note: 'New' declared here
  static Local<Integer> New(Isolate* isolate, int32_t value);
  ^
In file included from ../src/bufferutil.cc:15:
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:224:26: error: no member named 'New' in 'v8::String'
    obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
             ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:229:5: error: no member named 'ThrowException' in namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
      ~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:229:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:42: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                             ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:233:5: error: no member named 'ThrowException' in namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
      ~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:233:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:209:42: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                             ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:244:13: error: no member named 'smalloc' in namespace 'node'
      node::smalloc::FreeCallback callback,
      ~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:251:12: error: no matching function for call to 'New'
    return node::Buffer::New(data, size);
           ^~~~~~~~~~~~~~~~~
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:52:40: note: candidate function not viable: no known conversion from 'char *' to 'v8::Isolate *' for 1st argument
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:55:40: note: candidate function not viable: no known conversion from 'char *' to 'v8::Isolate *' for 1st argument
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:67:40: note: candidate function not viable: requires 3 arguments, but 2 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:60:40: note: candidate function not viable: requires 5 arguments, but 2 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
In file included from ../src/bufferutil.cc:15:
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:255:12: error: no matching function for call to 'New'
    return node::Buffer::New(size);
           ^~~~~~~~~~~~~~~~~
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:52:40: note: candidate function not viable: requires 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:55:40: note: candidate function not viable: requires at least 2 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:67:40: note: candidate function not viable: requires 3 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/ken/.node-gyp/8.11.3/include/node/node_buffer.h:60:40: note: candidate function not viable: requires 5 arguments, but 1 was provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
In file included from ../src/bufferutil.cc:15:
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:259:26: error: no member named 'Use' in namespace 'node::Buffer'
    return node::Buffer::Use(data, size);
           ~~~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:450:49: error: too few arguments to function call, single argument 'isolate' was not specified
    v8::Local<v8::Object> obj = v8::Object::New();
                                ~~~~~~~~~~~~~~~ ^
/Users/ken/.node-gyp/8.11.3/include/node/v8.h:3491:3: note: 'New' declared here
  static Local<Object> New(Isolate* isolate);
  ^
In file included from ../src/bufferutil.cc:15:
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:451:14: error: no member named 'NewSymbol' in 'v8::String'
    obj->Set(NanSymbol("callback"), fn);
             ^~~~~~~~~~~~~~~~~~~~~
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:103:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^
/Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/nan/nan.h:457:12: error: no member named 'Dispose' in 'v8::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
    handle.Dispose();
    ~~~~~~ ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/bufferutil/src/bufferutil.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/ken/Desktop/face-detection-node-opencv/server/node_modules/engine.io-client/node_modules/ws
gyp ERR! node -v v8.11.3
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
